### PR TITLE
fix #12293 findNimStdLibCompileTime should not break with nimble install compiler

### LINKED ISSUE #12293: compiler as a library: findNimStdLibCompileTime fails to find system.nim
When trying the compiler as library API, findNimStdLibCompileTime does not find system.nim when the compiler is installed through nimble.

Test case:

nimble install compiler

And then compile:
```Nim
import compiler/[nimeval, llstream]

proc evalString(code: string, moduleName = "script.nim") =
  let stream = llStreamOpen(code)
  let std = findNimStdLibCompileTime()
  var intr = createInterpreter(moduleName, [std])
  intr.evalScript(stream)
  destroyInterpreter(intr)
  llStreamClose(stream)

let runtimeString = "echo \"Hello World\""
evalString(runtimeString)
```
Error:
```
/home/beta/Programming/Nim/Arraymancer/build/ceval.nim(13) ceval
/home/beta/Programming/Nim/Arraymancer/build/ceval.nim(5) evalString
/home/beta/.nimble/pkgs/compiler-1.0.0/compiler/nimeval.nim(98) findNimStdLibCompileTime
/home/beta/.choosenim/toolchains/nim-1.0.0/lib/system/assertions.nim(27) failedAssertImpl
/home/beta/.choosenim/toolchains/nim-1.0.0/lib/system/assertions.nim(20) raiseAssert
/home/beta/.choosenim/toolchains/nim-1.0.0/lib/system/fatal.nim(39) sysFatal
Error: unhandled exception: /home/beta/.nimble/pkgs/compiler-1.0.0/compiler/nimeval.nim(98, 12) `fileExists(result / "system.nim")` result:/home/beta/.nimble/pkgs/compiler-1.0.0/lib [AssertionError]
```

from:
https://github.com/nim-lang/Nim/blob/af6539b0c3b6958251737d5bc140cbcc8652eb8b/compiler/nimeval.nim#L93-L98

### PR DESCRIPTION
**Pull request by timotheecour**
* fix #12293
* there are still some subtleties as mentioned in https://github.com/nim-lang/Nim/issues/12293#issuecomment-627958077
but this should work at least:

```
# with nimble develop
cd $nim
NIMBLE_DIR=$HOME/.nimble_fake24 nimble develop
# or, with nimble install for any version recent enough to have this patch:
NIMBLE_DIR=$HOME/.nimble_fake24 nimble install https://github.com/timotheecour/Nim@#pr_fix_findNimStdLibCompileTime

# then:
NIMBLE_DIR=$HOME/.nimble_fake24 nim r $timn_D/tests/nim/all/t10738.nim
```

```nim
# t10738.nim:
import compiler/[nimeval, llstream]
import os
proc evalString(code: string, moduleName = "script.nim") =
  let stream = llStreamOpen(code)
  let lib = findNimStdLibCompileTime()
  var intr = createInterpreter(moduleName, [lib, lib / "pure", lib / "core"])
  intr.evalScript(stream)
  destroyInterpreter(intr)
  llStreamClose(stream)

let runtimeString = "echo \"Hello World\""
evalString(runtimeString)
```

### CODE REVIEW

File: compiler/nimeval.nim [hunk 1]
```diff
@@ -93,8 +96,8 @@ proc findNimStdLib*(): string =
 proc findNimStdLibCompileTime*(): string =
   ## Same as ``findNimStdLib`` but uses source files used at compile time,
   ## and asserts on error.
-  const sourcePath = currentSourcePath()
-  result = sourcePath.parentDir.parentDir / "lib"
+  const exe = getCurrentCompilerExe()
+  result = exe.parentDir.parentDir / "lib"
```
Comment (Araq) [inline compiler/nimeval.nim]: This should be ``result = exe.splitFile.dir.parentDir / "lib"``

Comment (timotheecour) [reply compiler/nimeval.nim]: I know there are semantic differences bw parentDir and splitFile.dir (and IMO parentDir always has the more intuitive semantics; in fact splitFile still has some bugs right now), but why do you prefer splitFile.dir here? 

what's missing in runnableExamples for `parentDir` are tests for input filenames (eg "foo/bar.txt"); i can add an example + add tests + clarify it in docs for parentDir but this does work reliably (more reliably than splitFile for corner cases)

(parentDir does less work too but I'm not concerned with that here)

Comment (Araq) [reply compiler/nimeval.nim]: ``exe`` is a "full" filename and it has a dir component of which we select the parent dir. By your logic a filename's parentdir is the same as the directory it's in, that's confusing.

Comment (timotheecour) [reply compiler/nimeval.nim]: I've made your suggestion change to unblock this PR but will discuss this separately later

**Comment by timotheecour** [comment]
PTAL
~~CI failure unrelated: https://github.com/nim-lang/Nim/issues/14090~~ re-ran
